custom_metrics/Images.js

```diff
@@ -18,7 +18,11 @@ var wptImages = function(win) {
             naturalHeight: el.naturalHeight,
             loading: el.getAttribute("loading"),
             decoding: el.getAttribute("decoding"),
-            inViewport: el.getBoundingClientRect().top < window.innerHeight,
+            inViewport:
+              el.getBoundingClientRect().bottom >= 0 &&
+              el.getBoundingClientRect().right >= 0 &&
+              el.getBoundingClientRect().top <= window.innerHeight &&
+              el.getBoundingClientRect().left <= window.innerWidth,

```

Comment:
IIUC this only evaluates to `true` if the image is entirely contained within the viewport. Should it also evaluate to `true` if the image is partially in the viewport? Practically, do we also care about images that are peeking out above the fold like an iceberg because those are expected to be loaded eagerly?

Reply:
This evaluates to `true` if _any_ part of the image is in the viewport as we would expect that image to have been loaded and `loading=lazy` would load such images. I believe this to be more useful than the alternative below.

If we are only interested in images which are completely within the viewport, then we would change the above to:

```
inViewport:
  el.getBoundingClientRect().top >= 0 &&
  el.getBoundingClientRect().left >= 0 &&
  el.getBoundingClientRect().bottom <= window.innerHeight &&
  el.getBoundingClientRect().right <= window.innerWidth  
```

Reply:
Thanks for clarifying 😅